lib/IRGen/GenFunc.cpp

```diff
@@ -1301,7 +1331,12 @@ void irgen::emitFunctionPartialApplication(IRGenFunction &IGF,
   // to capture), and the dest ownership semantics match the parameter's,
   // skip building the box and thunk and just take the pointer as
   // context.
-  if (!origType->isPolymorphic() &&
+  // TODO: We can only do this and use swiftself if all our swiftcc emit the
+  // last parameter that fits into a register as swiftself.
+  // We should get this optimization back using the @convention(closure) whose
+  // box argument should just be swift self.
+  if (false &&
```

Comment:
This is resulting in a build warning about unreachable code.

Can you (ideally) remove this code and open a bug to restore the behavior that you'd like to eventually see, or alternately follows he advice in the note we emit (to add parens around false to silence the warning).